go/engine/account_reset.go

```diff
@@ -42,6 +42,61 @@ func (e *AccountReset) SubConsumers() []libkb.UIConsumer {
 }
 
 // Run starts the engine.
-func (e *AccountReset) Run(mctx libkb.MetaContext) error {
-	return libkb.EnterResetPipeline(mctx, e.username, e.email)
+func (e *AccountReset) Run(mctx libkb.MetaContext) (err error) {
+	mctx = mctx.WithLogTag("RST")
+	defer mctx.TraceTimed("Account#Run", func() error { return err })()
+
+	// User's with active devices cannot reset at all
+	if mctx.ActiveDevice().Valid() {
+		return libkb.ResetWithActiveDeviceError{}
+	}
+
+	// We can enter the reset pipeline with exactly one of these parameters
+	// set. We first attempt to establish a session for the user. Otherwise we
+	// send up an email or username.
+	var self bool
+	var username, email string
+
+	mctx = mctx.WithNewProvisionalLoginContext()
+	err = libkb.PassphraseLoginPrompt(mctx, e.usernameOrEmail, 3)
```

Comment:
how does this feel for the user? Do they get a prompt when they just told you they've lost the account? Some text on the prompt that says "if you don't know it, that's fine you can use email instead"

Maybe asking them whether they'd like to enter their passphrase or use an email would be better

Reply:
I'll change the prompt to be specific to this call so we can customize it further. it currently has cancel/close buttons that allow you to skip the prompt, but I agree that it could be clarified as optional. When entering a passphrase you have to provide a username or email (or you will be prompted for one) which is why I made the cli arg required